MekHQ/src/mekhq/campaign/parts/Part.java

```diff
@@ -1775,4 +1761,144 @@ public SimpleTechLevel getStaticTechLevel() {
         }
         return getTechAdvancement().getStaticTechLevel();
     }
+
+    public void fixupPartReferences(Map<Integer, Part> knownParts) {
+        if (parentPart instanceof PartRef) {
+            int id = parentPart.getId();
+            parentPart = knownParts.get(id);
+            if (parentPart == null) {
+                MekHQ.getLogger().error(PartRef.class,
+                    String.format("Part %d ('%s') references missing parent part %d",
+                        getId(), getName(), id));
+            }
+        }
+
+        for (int ii = childParts.size() - 1; ii >= 0; --ii) {
+            Part childPart = childParts.get(ii);
+            if (childPart instanceof PartRef) {
+                Part realPart = knownParts.get(childPart.getId());
+                if (realPart != null) {
+                    childParts.set(ii, realPart);
+                } else {
+                    MekHQ.getLogger().error(PartRef.class,
+                        String.format("Part %d ('%s') references missing child part %d",
+                            getId(), getName(), id));
+                    childParts.remove(ii);
+                }
+            }
+        }
+    }
+
+    public static class PartRef extends Part {
+        /**
+         *
+         */
+        private static final long serialVersionUID = 1L;
```

Comment:
Why 1L instead of an actual serialVersionUID?

Reply:
1 is a valid version UID.